src/stk/itk_bridge/itk_convert.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include <stk/image/volume.h>
+
+#include <itkImageIOBase.h>
+
+namespace stk {
+
+
+/*!
+ * \brief Convert an ITK type to the corresponding STK type.
+ */
+template<typename TPixelType>
+stk::Type itk2stk_type(void)
+{
+    return stk::type_id<typename itk::PixelTraits<TPixelType>::ValueType>::id();
+}
+
+
+/*!
+ * \brief Convert an ITK object into an STK object.
+ * \note The new object owns a deep copy of the image data.
+ */
+template<class ImageType>
+stk::Volume itk2stk(const typename ImageType::Pointer & image)
+{
+    auto const region = image->GetLargestPossibleRegion();
+    auto const size = region.GetSize();
+
+    const dim3 stk_size = {uint32_t(size[0]), uint32_t(size[1]), uint32_t(size[2])};
+    const size_t voxel_no = size[0] * size[1] * size[2];
+    const stk::Type stk_type = itk2stk_type<typename ImageType::PixelType>();
+
+    if (stk::Type_Unknown == stk_type) {
+        FATAL() << "Conversion error, unknown or unsupported image type";
+    }
+
+    stk::Volume volume (stk_size, stk_type);
+
+    // Copy image data
+    std::memcpy(volume.ptr(),
+                image->GetBufferPointer(),
+                sizeof (typename ImageType::PixelType) * voxel_no);
```

Comment:
Here (and the same in the other conversion) I have to double-check whether there can be issues due to alignment padding or similar.